v3.go

```diff
@@ -391,7 +391,7 @@ func (x *GoSNMP) unmarshalV3Header(packet []byte,
 		return 0, errors.New("error parsing SNMPV3 message ID: truncted packet")
 	}
 
-	if MsgFlags, ok := rawMsgFlags.(string); ok {
+	if MsgFlags, ok := rawMsgFlags.(string); ok && len(MsgFlags) > 0 {
```

Comment:
Avoid `panic: runtime error: index out of range [0] with length 0` in the following line when `MsgFlags` is empty.

Tested by ` panicUnmarshalV3HeaderFlagLen`